contracts/contracts/strategies/NativeStaking/ValidatorAccountant.sol

```diff
@@ -0,0 +1,266 @@
+    /// @param _ssvNetwork Address of the SSV Network contract
+    constructor(
+        address _wethAddress,
+        address _vaultAddress,
+        address _beaconChainDepositContract,
+        address _ssvNetwork
+    )
+        ValidatorRegistrator(
+            _wethAddress,
+            _vaultAddress,
+            _beaconChainDepositContract,
+            _ssvNetwork
+        )
+    {}
+
+    /// @notice set fuse interval values
+    function setFuseInterval(
+        uint256 _fuseIntervalStart,
+        uint256 _fuseIntervalEnd
+    ) external onlyGovernor {
+        require(
+            _fuseIntervalStart < _fuseIntervalEnd &&
+                _fuseIntervalEnd < 32 ether &&
+                _fuseIntervalEnd - _fuseIntervalStart >= 4 ether,
+            "incorrect fuse interval"
+        );
+
+        emit FuseIntervalUpdated(_fuseIntervalStart, _fuseIntervalEnd);
+
+        fuseIntervalStart = _fuseIntervalStart;
+        fuseIntervalEnd = _fuseIntervalEnd;
+    }
+
+    /* solhint-disable max-line-length */
+    /// This notion page offers a good explanation of how the accounting functions
+    /// https://www.notion.so/originprotocol/Limited-simplified-native-staking-accounting-67a217c8420d40678eb943b9da0ee77d
+    /// In short, after dividing by 32, if the ETH remaining on the contract falls between 0 and fuseIntervalStart,
+    /// the accounting function will treat that ETH as Beacon chain consensus rewards.
+    /// On the contrary, if after dividing by 32, the ETH remaining on the contract falls between fuseIntervalEnd and 32,
+    /// the accounting function will treat that as a validator slashing.
+    /// @notice Perform the accounting attributing beacon chain ETH to either full or partial withdrawals. Returns true when
+    /// accounting is valid and fuse isn't "blown". Returns false when fuse is blown.
+    /// @dev This function could in theory be permission-less but lets allow only the Registrator (Defender Action) to call it
+    /// for now.
+    /// @return accountingValid true if accounting was successful, false if fuse is blown
+    /* solhint-enable max-line-length */
+    function doAccounting()
+        external
+        onlyRegistrator
+        whenNotPaused
+        returns (bool accountingValid)
+    {
+        // pause the accounting on failure
+        accountingValid = _doAccounting(true);
+    }
+
+    // slither-disable-start reentrancy-eth
+    function _doAccounting(bool pauseOnFail)
+        internal
+        returns (bool accountingValid)
+    {
+        if (address(this).balance < consensusRewards) {
+            return _failAccounting(pauseOnFail);
+        }
+
+        // Calculate all the new ETH that has been swept to the contract since the last accounting
+        uint256 newSweptETH = address(this).balance - consensusRewards;
+        accountingValid = true;
+
+        // send the ETH that is from fully withdrawn validators to the Vault
+        if (newSweptETH >= MAX_STAKE) {
```

Comment:
Depending on how often and how many validators you plan to run, this condition could return true even if no validator has fully withdrawn.

Lido operates approximately 300,000 validators. If you manage 1% of that, or 3,000 validators, your base daily rewards would be around 7 ETH, excluding execution rewards, which could quickly add up to 32 in just a few days.

Reply:
The idea is to cap the number a validators a strategy can manage to 600 (19,200 ETH)

Reply:
We need to decide if we are going to enforce the 600 validator (19,200 ETH) limit in the strategy contract or not. If all stETH and rETH currently in the OETH Vault was convert to WETH and staked, that would be 20,188 ETH.